nova/src/ccs/mod.rs

```diff
@@ -72,6 +72,21 @@ pub struct CCSShape<G: CurveGroup> {
     pub cSs: Vec<(G::ScalarField, Vec<usize>)>,
 }
 
+impl<G: CurveGroup> fmt::Display for CCSShape<G> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "CCSShape {{ num_constraints: {}, num_vars: {}, num_io: {}, num_matrices: {}, num_multisets: {}, max_cardinality: {}, Ms: [_, {}], cSs: [_, {}] }}",
```

Comment:
nit `A: [_, {N}]` in R1CS means that A is a matrix with `N` non-zero elements. Here it's `Ms: [_, {number of M matrices}]`

Reply:
I mean this is mostly a gut check anyway, the most relevant size information is captured in the first three parameters